[C++ Generator] Avoid unused functions

We should avoid warnings in generated C++ code so that we can activate -Werror.

in StatechartEntryExitActions
```
java.lang.RuntimeException: Compilation failed (exit status 1):
StatechartEntryExitActions.cc:50:6: error: ‘void {anonymous}::exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue()’ defined but not used [-Werror=unused-function]
 void exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue(){
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
StatechartEntryExitActions.cc:35:6: error: ‘void {anonymous}::entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue()’ defined but not used [-Werror=unused-function]
 void entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue(){
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

--- a/test-plugins/org.yakindu.sct.generator.c.test/gtests/StatechartEntryExitActions/StatechartEntryExitActions.cc
+++ b/test-plugins/org.yakindu.sct.generator.c.test/gtests/StatechartEntryExitActions/StatechartEntryExitActions.cc
@@ -12,9 +12,7 @@ class StatechartEntryExitActions : public ::testing::Test
 public:
 	/* All operations from the SCTUnit test class. */
 	void entryActionsAreExecutedOnEnteringStatechart();
-	void entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue();
 	void exitActionsAreExecutedOnEnteringStatechart();
-	void exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue();
 	void setTimer(StatechartEntryAndExitActions* statechart, const sc_eventid evid, const sc_integer time_ms, const sc_boolean periodic);
 	void unsetTimer(StatechartEntryAndExitActions* handle, const sc_eventid evid);
 protected:
@@ -24,8 +22,6 @@ protected:
 
 static StatechartEntryExitActions * tc;
 
-void entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue();
-void exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue();
 
 void StatechartEntryExitActions::SetUp()
 {
@@ -51,27 +47,11 @@ void StatechartEntryExitActions::entryActionsAreExecutedOnEnteringStatechart()
 	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_x(&statechart)== 5l);
 	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_y(&statechart)== 3l);
 }
-void StatechartEntryExitActions::entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue()
-{
-	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_x(&statechart)== 0l);
-	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_y(&statechart)== 0l);
-	statechartEntryAndExitActionsIface_set_b(&statechart,false);
-	statechartEntryAndExitActions_enter(&statechart);
-	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_x(&statechart)== 2l);
-	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_y(&statechart)== 3l);
-}
 void StatechartEntryExitActions::exitActionsAreExecutedOnEnteringStatechart()
 {
 	statechartEntryAndExitActions_enter(&statechart);
 	statechartEntryAndExitActions_exit(&statechart);
 	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_x(&statechart)== 6l);
-	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_y(&statechart)== 2l);
-}
-void StatechartEntryExitActions::exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue()
-{
-	statechartEntryAndExitActions_enter(&statechart);
-	statechartEntryAndExitActions_exit(&statechart);
-	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_x(&statechart)== 8l);
 	EXPECT_TRUE(statechartEntryAndExitActionsIface_get_y(&statechart)== 2l);
 }
 

--- a/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/StatechartEntryExitActions/StatechartEntryExitActions.cc
+++ b/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/StatechartEntryExitActions/StatechartEntryExitActions.cc
@@ -6,8 +6,6 @@
 
 namespace  {
 
-void entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue();
-void exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue();
 statechartactions::StatechartEntryAndExitActions* statechart;
 
 
@@ -32,32 +30,6 @@ class StatechartEntryExitActions : public ::testing::Test{
 	}
 };
 
-void entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue(){
-	EXPECT_TRUE(statechart->getDefaultSCI()->get_x()== 0l);
-	
-	EXPECT_TRUE(statechart->getDefaultSCI()->get_y()== 0l);
-	
-	statechart->getDefaultSCI()->set_b(false);
-	
-	statechart->enter();
-	
-	EXPECT_TRUE(statechart->getDefaultSCI()->get_x()== 2l);
-	
-	EXPECT_TRUE(statechart->getDefaultSCI()->get_y()== 3l);
-	
-	
-}
-void exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue(){
-	statechart->enter();
-	
-	statechart->exit();
-	
-	EXPECT_TRUE(statechart->getDefaultSCI()->get_x()== 8l);
-	
-	EXPECT_TRUE(statechart->getDefaultSCI()->get_y()== 2l);
-	
-	
-}
 
 TEST_F(StatechartEntryExitActions, entryActionsAreExecutedOnEnteringStatechart) {
 	

--- a/test-plugins/org.yakindu.sct.test.models/tests/statechartactions/StatechartEntryExitActions.sctunit
+++ b/test-plugins/org.yakindu.sct.test.models/tests/statechartactions/StatechartEntryExitActions.sctunit
@@ -11,18 +11,6 @@ testclass StatechartEntryExitActions for statechart statechartactions.Statechart
 		assert y == 3 		
 	}
 
-// Unused - commented out
-//	operation entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue() {
-//		
-//		assert x == 0
-//		assert y == 0
-//		b = false
-//		
-//		enter
-//		assert x == 2 
-//		assert y == 3 		
-//	}
-
 	@Test operation exitActionsAreExecutedOnEnteringStatechart() {
 
 		enter		
@@ -30,15 +18,4 @@ testclass StatechartEntryExitActions for statechart statechartactions.Statechart
 		assert x == 6
 		assert y == 2 		
 	}
-
-// Unused - commented out
-//	operation exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue() {
-//		
-//		enter		
-//		exit
-//		assert x == 8
-//		assert y == 2 		
-//	}
-
-
 }

--- a/test-plugins/org.yakindu.sct.test.models/tests/statechartactions/StatechartEntryExitActions.sctunit
+++ b/test-plugins/org.yakindu.sct.test.models/tests/statechartactions/StatechartEntryExitActions.sctunit
@@ -11,17 +11,17 @@ testclass StatechartEntryExitActions for statechart statechartactions.Statechart
 		assert y == 3 		
 	}
 
-
-	operation entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue() {
-		
-		assert x == 0
-		assert y == 0
-		b = false
-		
-		enter
-		assert x == 2
-		assert y == 3 		
-	}
+// Unused - commented out
+//	operation entryActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue() {
+//		
+//		assert x == 0
+//		assert y == 0
+//		b = false
+//		
+//		enter
+//		assert x == 2 
+//		assert y == 3 		
+//	}
 
 	@Test operation exitActionsAreExecutedOnEnteringStatechart() {
 
@@ -31,13 +31,14 @@ testclass StatechartEntryExitActions for statechart statechartactions.Statechart
 		assert y == 2 		
 	}
 
-	operation exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue() {
-		
-		enter		
-		exit
-		assert x == 8
-		assert y == 2 		
-	}
+// Unused - commented out
+//	operation exitActionsAreExecutedOnEnteringStatechartOnlyIfGuardIsTrue() {
+//		
+//		enter		
+//		exit
+//		assert x == 8
+//		assert y == 2 		
+//	}
 
 
 }

